InscryptionAPI/Ascension/StarterDeckPaginator.cs

```diff
@@ -1,104 +1,102 @@
-            icon.gameObject.SetActive(false);
-
-        // Start going through and setting the icons
-        for (int i = 0; i < visibleStarterDecks.Count; i++)
+        pageIndex++;
+        if (pageIndex >= pages.Count)
         {
-            if (visibleStarterDecks[i] == null) // this is a spacer
-                continue;
-            AscensionStarterDeckIcon targetIcon = this.icons[i];
-            targetIcon.AssignInfo(visibleStarterDecks[i]);
-            targetIcon.gameObject.SetActive(true);
+            pageIndex = 0;
         }
+        LoadPage(pages[pageIndex]);
     }
 
-    public void StarterDeckPageLeft(MainInputInteractable button)
+    public void PreviousPage()
     {
-        if (!AscensionMenuScreens.Instance.starterDeckSelectScreen.activeSelf)
-            return;
-
-        if (starterDeckPageIndex > 0)
+        pageIndex--;
+        if (pageIndex < 0)
         {
-            starterDeckPageIndex -= 1;
-            ShowVisibleStarterDecks();
+            pageIndex = pages.Count - 1;
         }
+        LoadPage(pages[pageIndex]);
     }
 
-    public void StarterDeckPageRight(MainInputInteractable button)
+    public void LoadPage(List<StarterDeckInfo> page)
     {
-        if (!AscensionMenuScreens.Instance.starterDeckSelectScreen.activeSelf)
-            return;
-
-        if (starterDeckPageIndex < pages.Count - 1)
+        List<AscensionStarterDeckIcon> sorted = new List<AscensionStarterDeckIcon>(screen.deckIcons);
+        sorted.Sort((x, x2) => Mathf.RoundToInt((Mathf.Abs(x.transform.position.y - x2.transform.position.y) < 0.1f ? x.transform.position.x - x2.transform.position.x : x2.transform.position.y - x.transform.position.y) * 100));
+        for (int i = 0; i < pageLength; i++)
         {
-            starterDeckPageIndex += 1;
-            ShowVisibleStarterDecks();
+            if (i < sorted.Count)
+            {
+                if (i < page.Count)
+                {
+                    sorted[i].starterDeckInfo = page[i];
+                    sorted[i].AssignInfo(page[i]);
+                    if (!sorted[i].Unlocked)
+                    {
+                        sorted[i].conqueredRenderer.enabled = false;
+                    }
+                }
+                else
+                {
+                    sorted[i].starterDeckInfo = null;
+                    sorted[i].AssignInfo(null);
+                    sorted[i].conqueredRenderer.enabled = false;
+                    sorted[i].iconRenderer.sprite = noneDeckSprite;
+                }
+            }
+        }
+        noneDeckSprite ??= TextureHelper.GetImageAsTexture("starterdeck_icon_none.png", Assembly.GetExecutingAssembly()).ConvertTexture();
+        foreach (AscensionStarterDeckIcon icon in screen.deckIcons)
+        {
+            if (icon.Info == null)
+                icon.iconRenderer.sprite = noneDeckSprite;
         }
+        CommandLineTextDisplayer.PlayCommandLineClickSound();
     }
 
-    public void OnEnable()
```

Comment:
This code was actually really important for 'total conversion' mods like P03. It allowed mods to hook into an event that fired _right before_ the starter deck screen loaded, and let them modify starter decks based on screen context, and have those changes immediately reflected in the screen.

Even if the use case is relatively narrow, it's a breaking change, and I don't see any reason not to include this logic in the new version. Just manually resync the deck lists and regenerate the pages each time the screen loads.

Reply:
ok

Reply:
should hopefully work